ui/pages/settings/networks-tab/networks-form/networks-form.js

```diff
@@ -488,39 +571,29 @@ const NetworksForm = ({
         />
         <FormField
           error={errors.rpcUrl?.msg || ''}
-          onChange={(value) => {
-            setRpcUrl(value);
-            validateUrlRpcUrl(value);
-          }}
+          onChange={setRpcUrl}
           titleText={t('rpcUrl')}
           value={rpcUrl}
           disabled={viewOnly}
         />
         <FormField
           error={errors.chainId?.msg || ''}
-          onChange={(value) => {
-            setChainId(value);
-            validateChainIdOnChange(value);
-          }}
+          onChange={setChainId}
           titleText={t('chainId')}
           value={chainId}
           disabled={viewOnly}
           tooltipText={viewOnly ? null : t('networkSettingsChainIdDescription')}
         />
         <FormField
-          error={errors.ticker?.msg || ''}
+          warning={warnings.ticker?.msg || ''}
           onChange={setTicker}
           titleText={t('currencySymbol')}
-          titleUnit={t('optionalWithParanthesis')}
           value={ticker}
           disabled={viewOnly}
         />
         <FormField
           error={errors.blockExplorerUrl?.msg || ''}
-          onChange={(value) => {
-            setBlockExplorerUrl(value);
-            validateBlockExplorerURL(value);
-          }}
+          onChange={setBlockExplorerUrl}
           titleText={t('blockExplorerUrl')}
           titleUnit={t('optionalWithParanthesis')}
           value={blockExplorerUrl}
```

Comment:
@adonesky1 : this file is getting too big, it will be useful to extract validation logic into a separate file and have more unit test coverage.

Reply:
One problem with this is that the validation methods require the context from `useI18nContext` and it feels wrong to pass it as an argument? And I think we agreed that we shouldn't pass keys to `t()` (from`useI18nContext`) as variables. I suppose I could make a class that holds these methods that instantiates `useI18nContext`, and is itself instantiated in the networks-form component but that isn't a pattern I think we use anywhere else so I'm hesitant. Do you have thoughts on how we should create helper files that need context like `useI18nContext` without passing it in as an argument? cc @jpuri @brad-decker @danjm

Reply:
I remember @brad-decker also raising whether we should deprecate the recently introduced `<I18nValue>` what was the result of that discussion?

Reply:
@jpuri added more unit testing here: https://github.com/MetaMask/metamask-extension/pull/12431/commits/ba7b38dbf276a2bfa550fcd5907e5c06deca9dfd